app/assets/javascripts/discourse/app/widgets/home-logo.js

```diff
@@ -17,57 +17,119 @@ export default createWidget("home-logo", {
     return typeof href === "function" ? href() : href;
   },
 
-  logoUrl() {
-    return this.siteSettings.site_logo_url || "";
+  logoUrl(opts = {}) {
+    return this.logoResolver("logo", opts);
   },
 
-  mobileLogoUrl() {
-    return this.siteSettings.site_mobile_logo_url || "";
+  mobileLogoUrl(opts = {}) {
+    return this.logoResolver("mobile_logo", opts);
   },
 
-  smallLogoUrl() {
-    return this.siteSettings.site_logo_small_url || "";
+  smallLogoUrl(opts = {}) {
+    return this.logoResolver("logo_small", opts);
   },
 
   logo() {
-    const { siteSettings } = this;
-    const mobileView = this.site.mobileView;
+    const { siteSettings } = this,
+      mobileView = this.site.mobileView,
+      darkModeOn = document.head.querySelectorAll(
```

Comment:
I think it might be a good idea to either add this to our `session` or `caps` objects when the app boots up. Then the widget needn't rely on the document.